### DISCUSSION TITLE
Return the current role id of the owner with an item

### DISCUSSION THREAD

**Discussion by RobinJespersen** [directus/directus#2790] [Feature Requests]
Problem:
We have a custom frontend that has to decide if an action button related to an item is shown or not.

We know the Id of the user and also the permissions and we know the id of the owner of the item.
So deciding whether the action button is shown or not is possible for none, mine, and full, but not for role.

Solution:
For each collection, if an owner column is present, return an additional column with the current role id of the owner.

**Comment by rijkvanzanten** [comment]
> For each collection, if an owner column is present, return an additional column with the current role id of the owner.

Have you tried fetching the role relationally nested?

`GET /items/:collection?fields=*,owner.id,owner.role` should return the item with the role field of the owner included.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
